tools/yaml-templates/ios-test.yml

```diff
@@ -148,7 +151,7 @@ steps:
       echo "ls -lR ${output}"
       ls -lR "${output}"
     displayName: Preparations for publishing results
-    condition: always()
+    condition: eq(variables['xctestHtmlReportInstalled'], 'Yes')
```

Comment:
What effect will this have? Will the test results not be published if installation failed? Does that mean that any test failures will be overlooked if the html reporter didn't get installed?

Reply:
This condition is to say, only when `xctestHtmlReportInstall` task successfully be executed, this task will then be executed to publish results to the artifact.

Any test failures will NOT be overlooked. It just affects devs to figure out why test cases fail on CI pipeline with detailed screenshots and logs.

Reply:
Can you add a comment explaining that? Thanks!

Reply:
just wanted to make sure we are on the same page:

This does look like improvement, i.e, if we don't have `xctestHtmlReport` tool installed correctly, we shouldn't try to use the tool in later phase. The PR is not fixing existing environment issue, but rather than a pipeline task improvement. is my understanding correct ? @KangxuanYe

Reply:
Yes, 1000 out of 100 @maglims 

@AE-MS I'll add comment to explain the change over tasks.

Reply:
comments added.